File: 107_Binary Tree Level Order Traversal II/levelOrderBottom.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> list = new ArrayList<List<Integer>>();
        if(root == null)
            return list;
        
        Deque<TreeNode> deque = new ArrayDeque<>();
        deque.add(root);
        int curLay = 1;
        int nextLay = 0;
        
        List<Integer> curList = new ArrayList<>();
        while(!deque.isEmpty())
        {
            TreeNode node = deque.pop();
            curList.add(node.val);
            curLay--;
            
            if(node.left != null) {
                deque.add(node.left);
                nextLay++;
            }
            if(node.right != null) {
                deque.add(node.right);
                nextLay++;
            }
            
            if(curLay == 0)
            {
                curLay = nextLay;
                nextLay = 0;
                list.add(curList);
                curList = new ArrayList<>();
            }
        }
        
        return list;
    }
}

File: 107_BinaryTreeLevelOrderTraversal_II/levelOrderBottom.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

from collections import deque


class Solution:
    def levelOrderBottom(self, root: TreeNode) -> List[List[int]]:
        output = []
        cur_nodes = deque()
        cur_nodes.append(root)
        while cur_nodes:
            nex_nodes = deque()
            cur_val = []
            for nd in cur_nodes:
                if nd:
                    cur_val.append(nd.val)
                    nex_nodes.extend([nd.left, nd.right])
            if cur_val:
                output.insert(0, cur_val)
            cur_nodes = nex_nodes
        return output
